src/state/digital-exchange/actions.js

```diff
@@ -83,20 +90,45 @@ export const filterByRating = (rating, paginationMetadata) => {
   return applyFilter(filter, paginationMetadata);
 };
 
+export const resetSearchFilter = paginationMetadata => (dispatch, getState) => {
+  const selectedCategory = getSelectedDECategory(getState());
+  dispatch(setDEFilter(
+    { formValues: { name: null }, operators: { name: null } },
+    selectedCategory,
+  ));
+  dispatch(setDEFilter(
+    { formValues: { description: null }, operators: { description: null } },
+    selectedCategory,
+  ));
+  dispatch(setDEFilter({ formValues: { id: null }, operators: { id: null } }, selectedCategory));
+  dispatch(setDEFilter(
+    { formValues: { version: null }, operators: { version: null } },
+    selectedCategory,
+  ));
+  return dispatch(fetchDEComponents(
+    paginationMetadata,
+    genFilterParams(selectedCategory, getState),
+  ));
+};
+
 export const filterBySearch = (keyword, paginationMetadata) => {
-  const filter = {
-    formValues: {
-      name: keyword,
-      description: keyword,
-      id: keyword,
-      version: keyword,
-    },
-    operators: {
-      name: FILTER_OPERATORS.LIKE,
-      description: FILTER_OPERATORS.LIKE,
-      version: FILTER_OPERATORS.LIKE,
-      id: FILTER_OPERATORS.LIKE,
-    },
-  };
-  return applyFilter(filter, paginationMetadata);
+  let filter = null;
```

Comment:
Sorry, we normally don't allow using `let` declaration.  But I would suggest to use ternary operation for this entire function.